redis/src/types.rs

```diff
@@ -1350,7 +1350,93 @@ pub trait RedisWrite {
 
     /// Appends an empty argument to the command, and returns a
     /// [`std::io::Write`] instance that can write to it.
+    ///
+    /// Writing multiple arguments into this buffer is unsupported. The resulting
+    /// data will be interpreted as one argument by Redis.
+    ///
+    /// Writing no data is supported and is similar to having an empty bytestring
+    /// as an argument.
     fn writer_for_next_arg(&mut self) -> impl std::io::Write + '_;
+
+    /// Reserve space for `additional` arguments in the command
+    ///
+    /// `additional` is a list of the byte sizes of the arguments.
+    ///
+    /// It's strongly recommended that implementors implement this function,
+    /// as the default implementation is a no-op.
+    fn reserve_space_for_args(&mut self, additional: &[usize]) {
```

Comment:
please provide an example for when this can be used.

Reply:
I've added an example, and am thinking to change the function signature of `reserve_space_for_args` to:
```rust
fn reserve_space_for_args(&mut self, additional: impl Itererator<Item = usize>);
```

Reply:
yes, that makes sense to me.

Reply:
Done